zookeeper-api/src/main/java/org/apache/helix/zookeeper/impl/client/FederatedZkClient.java

```diff
@@ -479,8 +479,18 @@ public long getCreationTime(String path) {
 
   @Override
   public List<OpResult> multi(Iterable<Op> ops) {
-    throwUnsupportedOperationException();
-    return null;
+    if (ops == null) {
+      throw new NullPointerException("ops must not be null.");
+    }
+    String op_path = ops.iterator().next().getPath();
```

Comment:
I think we can call `getZkRealm(op_path)` here and store the realm to same some map lookup.